include/grpcpp/security/tls_credentials_options.h

```diff
@@ -0,0 +1,260 @@
+    void* config_user_data, grpc_tls_server_authorization_check_arg* arg);
+void tls_server_authorization_check_config_c_cancel(
+    void* config_user_data, grpc_tls_server_authorization_check_arg* arg);
+
+/** TLS server authorization check config, wraps
+ *  grps_tls_server_authorization_check_config. **/
+class TlsServerAuthorizationCheckConfig {
+ public:
+  TlsServerAuthorizationCheckConfig(
+      const void* config_user_data,
+      int (*schedule)(void* config_user_data,
+                      TlsServerAuthorizationCheckArg* arg),
+      void (*cancel)(void* config_user_data,
+                     TlsServerAuthorizationCheckArg* arg),
+      void (*destruct)(void* config_user_data));
+  ~TlsServerAuthorizationCheckConfig();
+
+  int Schedule(TlsServerAuthorizationCheckArg* arg) const {
+    return schedule_(config_user_data_, arg);
+  }
+
+  void Cancel(TlsServerAuthorizationCheckArg* arg) const {
+    if (cancel_ == nullptr) {
+      gpr_log(GPR_ERROR, "cancel API is nullptr");
+      return;
+    }
+    cancel_(config_user_data_, arg);
+  }
+
+  /** Creates C struct for the credential reload config. **/
+  grpc_tls_server_authorization_check_config* c_server_authorization_check()
+      const {
+    return c_config_;
+  }
+
+ private:
+  grpc_tls_server_authorization_check_config* c_config_;
+  void* config_user_data_;
+  int (*schedule_)(void* config_user_data, TlsServerAuthorizationCheckArg* arg);
+  void (*cancel_)(void* config_user_data, TlsServerAuthorizationCheckArg* arg);
+  void (*destruct_)(void* config_user_data);
+};
+
+/** TLS credentials options, wrapper for grpc_tls_credentials_options. **/
+class TlsCredentialsOptions {
+ public:
+  /** Getters for member fields. **/
+  grpc_ssl_client_certificate_request_type cert_request_type() const {
+    return cert_request_type_;
+  }
+  std::shared_ptr<TlsKeyMaterialsConfig> key_materials_config() const {
+    return key_materials_config_;
+  }
+  std::shared_ptr<TlsCredentialReloadConfig> credential_reload_config() const {
+    return credential_reload_config_;
+  }
+  std::shared_ptr<TlsServerAuthorizationCheckConfig>
+  server_authorization_check_config() const {
+    return server_authorization_check_config_;
+  }
+
+  /** Setters for member fields. **/
+  void set_cert_request_type(
+      const grpc_ssl_client_certificate_request_type type) {
+    cert_request_type_ = type;
+  }
+  void set_key_materials_config(std::shared_ptr<TlsKeyMaterialsConfig> config) {
+    key_materials_config_ = std::move(config);
+  }
+  void set_credential_reload_config(
+      std::shared_ptr<TlsCredentialReloadConfig> config) {
+    credential_reload_config_ = std::move(config);
+  }
+  void set_server_authorization_check_config(
+      std::shared_ptr<TlsServerAuthorizationCheckConfig> config) {
+    server_authorization_check_config_ = std::move(config);
+  }
+
+  /** Creates C struct for TLS credential options. **/
+  grpc_tls_credentials_options* c_credentials_options() const;
```

Comment:
Why not make C struct of TLS credentials options as a private member field of `TlsCredentialsOptions`? Like what we did for other configs?

Reply:
I did it this way because the user will create instances of TlsKeyMaterialsConfig, TlsCredentialReloadConfig, and TlsServerAuthorizationCheckConfig (even if, under the hood, two of these are nothing more than the C configs). To not add the extra step of converting from C++ to C and for easier memory management, I made the TlsCredentialsOptions own smart pointers to exactly what the user created.

If you still feel like it would be better to just have a single private member field of type grpc_tls_credentials_options, then I can do it - it would just involve adding a constructor that takes in the other 3 configs. But let me know your thoughts.